### PR TITLE
 gcc: Simplify path appending for Linuxbrew

### PR DESCRIPTION
**Pull request by iMichka**

### CODE REVIEW

**Comment by sjackman** [comment]
I don't know of any distributions that use `/usr/lib32`. Can you give an example? I'd be inclined to remove it.

**Comment by iMichka** [comment]
I am sure I already saw that somewhere. Maybe it was ubuntu 10.04 ...

The spec say (https://wiki.ubuntu.com/MultiarchSpec) that they now use triplets. So even `/usr/lib64` is not valid anymore. It should be:
> /usr/lib/i386-linux-gnu
> /usr/lib/x86_64-linux-gnu

**Comment by sjackman** [comment]
Debian and Ubuntu have always used triplets. Redhat/Fedora use `/usr/lib` for 32-bit and `/usr/lib64` for 64-bit.

**Comment by iMichka** [comment]
I updated the paths to `/usr/lib64:/usr/lib/x86_64-linux-gnu:/usr/lib/i386-linux-gnu/:/usr/lib`. It should cover the most common cases.

**Comment by xu-cheng** [comment]
May I suggest to avoid the hardcoded `x86_64-linux-gnu`? You can get it by running `gcc -print-multiarch`.

**Comment by xu-cheng** [comment]
Also, we can adopt a similar approach to eliminate the hard coded `/usr/lib64/libstdc++.so.6` and `/lib64/libgcc_s.so.1`. So instead of having
```ruby
ln_s ["/usr/lib64/libstdc++.so.6", "/lib64/libgcc_s.so.1"], lib
```

It may be better to have:
```ruby
ln_s Utils.popen_read(ENV.cc, "-print-file-name=libstdc++.so.6").strip, lib
ln_s Utils.popen_read(ENV.cc, "-print-file-name=libgcc_s.so.1").strip, lib
```

A nice bonus is that it will work for ARM 32 as well.

**Comment by iMichka** [comment]
@xu-cheng Thank you for finding that out. Nice solution. It will be way more general :)

File: Formula/gcc.rb [hunk 1]
```diff
@@ -120,7 +120,8 @@ def install
       # Fix for system gccs that do not support -static-libstdc++
       # gengenrtl: error while loading shared libraries: libstdc++.so.6
       mkdir_p lib
-      ln_s ["/usr/lib64/libstdc++.so.6", "/lib64/libgcc_s.so.1"], lib
+      ln_s Utils.popen_read(ENV.cc, "-print-file-name=libstdc++.so.6").strip, lib
+      ln_s Utils.popen_read(ENV.cc, "-print-file-name=libgcc_s.so.1").strip, lib
```
Comment (sjackman) [inline Formula/gcc.rb]: Before it was always the host's libraries. Now it may be brewed libraries.
Should this be `/usr/bin/cc` rather than `ENV.cc`?
What happens when `ENV.cc` is `clang`?

File: Formula/gcc.rb [hunk 2]
```diff
@@ -140,13 +141,7 @@ def install
...
         # Set the search path for glibc libraries and objects, using the system's glibc
         # Fix the error: ld: cannot find crti.o: No such file or directory
-        # Order is important, we put the /usr/libXY first in the path
-        ENV.prepend_path "LIBRARY_PATH", "/usr/lib"
-        if Hardware::CPU.is_32_bit?
-          ENV.prepend_path "LIBRARY_PATH", "/usr/lib32"
-        else
-          ENV.prepend_path "LIBRARY_PATH", "/usr/lib64"
-        end
+        ENV.prepend_path "LIBRARY_PATH", Pathname.new(Utils.popen_read(ENV.cc, "-print-file-name=crti.o")).parent
```
Comment (sjackman) [inline Formula/gcc.rb]: Should this be `/usr/bin/cc` rather than `ENV.cc`?

File: Formula/gcc.rb (hunk 1 above)
Comment (xu-cheng) [reply Formula/gcc.rb]: > Before it was always the host's libraries. Now it may be brewed libraries.

But will this cause any trouble? I don't think so.

> Should this be /usr/bin/cc rather than ENV.cc?

I think it should be `ENV.cc` which is the actual compiler to build current formula. Further, /usr/bin/cc may not be guaranteed to exist.

> What happens when ENV.cc is clang?

My test shows it works as intended.

**Comment by iMichka** [comment]
Let's go with that version. I am fine with `ENV.cc`. If more edge cases pop up we can revise this later, but it should work in more cases than before.
